### ISSUE TITLE
Opening a file with `HDF5Widget` locks the file

### ISSUE THREAD

**Issue by reszelaz** [vasole/pymca#822]
Hi,
This is a question on how to properly use the `HDF5Widget`. In Sardana we show this widget from the Spock CLI (IPython).
While the widget is being showed it is not possible to perform the scan. On Sardana side currently we can either:
- open the file once with `swmr_mode = True` for a multi scan session
- open and close if for each scan 

always with `r+` mode (unless the file is opened for the first time, then we use `w-` mode).

When I try to scan while the `HDF5Widget` is being showed, then I get:
```
IOError: Unable to open file (Unable to flock file, errno = 11, error message = 'resource temporarily unavailable')
```
This can be avoided by setting `HDF5_USE_FILE_LOCKING="FALSE"` environment variable for the `HDF5Widget` process, which in our case is the whole Spock process. 

Is there any other way to avoid this problem?

**Comment by vasole** [comment]
In our case readers and writers are separated...

The cleanest way to solve the issue would be to use a recent version of h5py and HDF5 1.12.1 The latter added support for using locking or not when opening the file.

https://github.com/h5py/h5py/pull/1952

**Comment by vasole** [comment]
HDF5Widget also accepts a weak reference to the file thus not having ownership of it. The naming should be changed because a phynxFile is nothing else than an h5py object, but the method to look at is:

`appendPhynxFile(self, phynxFile, weakreference=True)`

If everything works as expected, the writer should be able to do:

```
fileModel = FileModel()
fileView = HDF5Widget(fileModel)
phynxFile = fileModel.openFile(h5py_file_instance)
```

**Comment by vasole** [comment]
Sorry, I had to update the code fragment.

**Comment by reszelaz** [comment]
Hi @vasole,

Many thanks for your hints!

I tried `appendPhynxFile()` and it works like a charm when when using the example code from the main section of the `HDF5Widget` module. When I tried it with our Taurus/Sardana code I found some problems but it must be due to some special usage we do.

Making further tests I have discovered that I won't need to use the `appendPhynxFile()` approach. It is fine to open the file in SWMR mode by the writer. Then I don't get the "Unable to flock file" errors when opening the file with `HDF5Widget` and I can continue to write to the file with the SWMR writer while the widget is opened. So, for the current needs this is enough.

> In our case readers and writers are separated...

In our case this is the same. I was not clear enough in the issue description. So, we have MacroServer which is the writer and Spock (IPython) and its `showscan` magic command which shows the `HDF5Widget` is the reader.

> The cleanest way to solve the issue would be to use a recent version of h5py and HDF5 1.12.1 The latter added support for using locking or not when opening the file.

Good to know. Thanks for pointing to this!

From my side, there are no more doubts. I'll proceed to close.
